jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/StandardWarExplodedProcessor.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.war;
+
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.cli.ArtifactLayers;
+import com.google.cloud.tools.jib.cli.ArtifactProcessor;
+import com.google.cloud.tools.jib.plugins.common.ZipUtil;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.MoreFiles;
+import com.google.common.io.RecursiveDeleteOption;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+class StandardWarExplodedProcessor implements ArtifactProcessor {
+
+  private final Path warPath;
+  private final Path targetExplodedWarRoot;
+  private final Integer warJavaVersion;
+  private final AbsoluteUnixPath appRoot;
+
+  /**
+   * Constructor for {@link StandardWarExplodedProcessor}.
+   *
+   * @param warPath path to WAR file
+   * @param targetExplodedWarRoot path to exploded-war root
+   * @param warJavaVersion war java version
+   * @param appRoot the absolute path of the app on the container
```

Comment:
We don't provide the `--appRoot` option yet, right?

Reply:
Ah good question. Nope, we don't provide the `--appRoot` option yet, it will be added when `War.class` is implemented. This class will only accept appRoot as an input but the idea is to set it in `ArtifactProcessors`.  If a base image is specified then the user is expected to provide the `appRoot` (otherwise we throw an exception). If a base image is not specified then we create `StandardWarExplodedProcessor` with the default jetty app root.

Reply:
Sounds like a good plan.

> If a base image is specified then the user is expected to provide the `appRoot` (otherwise we throw an exception).

Maybe throw an exception only when it's not the `jetty` on Docker Hub. I imagine often you set a base image only to use a tag or digest for `jetty`, which was the case in #3204.

Reply:
Yes, that is a great point! Will have to keep this in mind when implementing `ArtifactProcessors`